MediaBrowser.Controller/Entities/BaseItem.cs

```diff
@@ -801,6 +801,43 @@ public virtual string CreatePresentationUniqueKey()
             return Id.ToString("N", CultureInfo.InvariantCulture);
         }
 
+        public virtual bool CanMove()
+        {
+            if (SourceType == SourceType.Channel)
+            {
+                return ChannelManager.CanMove(this);
+            }
+
+            return IsFileProtocol;
```

Comment:
Correct me if I'm wrong. This would means that the user is allowed to move a single item in a library out of the library root path, and move a children like an episode out of the season path. If that is the case, this would not work well with our current library scanning code and those items may be deleted on next scan. Our current library scanning logic relies on the file hierarchy heavily and cannot support the flexible item sources like such movement could create.